### ISSUE TITLE
Allow multiple consumers per channel 

### ISSUE THREAD

**Issue by fsantagostinobietti** [smallrye/smallrye-reactive-messaging#2365]
For each incoming Smallrye messaging channel there is one consumer created. It processes one message at a time.

Is it possible to allow creation of multiple concurrent consumer clones (each with its own connection or connection pool) ?
This way, if we don't care about ordering, it would be possible to increase throughput a lot.

I see similar requests have been addressed for [Kafka](https://github.com/smallrye/smallrye-reactive-messaging/issues/329) and [RabbitMQ](https://github.com/smallrye/smallrye-reactive-messaging/issues/1699) with ad-hoc solutions.  Is it reasonable to have a more general solution also available for AMQP e JMS connectors?

Actually the same result can be obtained by manually (copy&paste) creating multiple channels config and consumer methods such as:

```properties
# Inbound
mp.messaging.incoming.request-ch0.connector=smallrye-jms
mp.messaging.incoming.request-ch0.destination=DEV.QUEUE.1
mp.messaging.incoming.request-ch0.session-mode=CLIENT_ACKNOWLEDGE
...
mp.messaging.incoming.request-ch1.connector=smallrye-jms
mp.messaging.incoming.request-ch1.destination=DEV.QUEUE.1
mp.messaging.incoming.request-ch1.session-mode=CLIENT_ACKNOWLEDGE

# Outbound
mp.messaging.outgoing.response-ch0.connector=smallrye-jms
mp.messaging.outgoing.response-ch0.destination=DEV.QUEUE.2
...
mp.messaging.outgoing.response-ch1.connector=smallrye-jms
mp.messaging.outgoing.response-ch1.destination=DEV.QUEUE.2
```

```java
@ApplicationScoped
public class MsgConsumer {

    @Inject
    @RestClient
    RemoteService remoteService;

    @Incoming("request-ch0")
    @Outgoing("response-ch0")
    public Uni<Message<String>> processMsgCh0(Message<String> msg) {
        return remoteService.async(msg.getPayload())
                .map(resp ->  Message.of(resp, msg.getAck() ));
    }

    @Incoming("request-ch1")
    @Outgoing("response-ch1")
    public Uni<Message<String>> processMsgCh1(Message<String> msg) {
        return remoteService.async(msg.getPayload())
                .map(resp ->  Message.of(resp, msg.getAck() ));
    }
}
```
It works (tested with jms connector) but it's a cumbersome workaround and does not scale well.

**Comment by cescoffier** [comment]
@ozangunalp just pushed a feature that is related.

**Comment by ozangunalp** [comment]
Yes this is related to #2382, documentation in  https://smallrye.io/smallrye-reactive-messaging/4.12.0/concepts/incoming-concurrency/ Although I've not tested with JMS connector, not sure if it'll improve/change things, as it dispatches messages on a thread pool.

**Comment by fsantagostinobietti** [comment]
@ozangunalp I've tested on JMS connector and it works fine. Consumers are working concurrently as expected.

Thread pool is not shared by consumers. It looks like every consumer has its own thread pool. But I think this is correct.

**Comment by ozangunalp** [comment]
Closed by #2382
